channel/consensus_channel/consensus_channel.go

```diff
@@ -428,3 +462,47 @@ func (v Vars) AsState(fp state.FixedPart) state.State {
 func (c *ConsensusChannel) Participants() []types.Address {
 	return c.fp.Participants
 }
+
+// CreateFromDirectFundingObjective accepts an objective and generates a new consensus channel from it.
+// It assumes that EVERY DirectFundingObjective is for a ledger channel.
+func CreateFromDirectFundingObjective(dfo directfund.Objective) (*ConsensusChannel, error) {
+	// The current assumption is that ANY direct funding objective is for a ledger channel
+	ledger := dfo.C
+
+	if !ledger.PostFundComplete() {
+		return nil, fmt.Errorf("expected funding for channel %s to be complete", dfo.C.Id)
+	}
+	signedPostFund := ledger.SignedPostFundState()
+	leaderSig, err := signedPostFund.GetParticipantSignature(uint(leader))
+	if err != nil {
+		return nil, fmt.Errorf("could not get leader signature: %w", err)
+	}
+	followerSig, err := signedPostFund.GetParticipantSignature(uint(follower))
+	if err != nil {
+		return nil, fmt.Errorf("could not get follower signature: %w", err)
+	}
+	signatures := [2]state.Signature{leaderSig, followerSig}
+
+	if len(signedPostFund.State().Outcome) != 1 {
+		return nil, fmt.Errorf("a consensus channel only supports a single asset")
+	}
+	assetExit := signedPostFund.State().Outcome[0]
+	turnNum := signedPostFund.State().TurnNum
+	outcome := FromExit(assetExit)
+
+	if ledger.MyIndex == uint(leader) {
+		con, err := NewLeaderChannel(ledger.FixedPart, turnNum, outcome, signatures)
+		if err != nil {
+			return nil, fmt.Errorf("could not create consensus channel as leader: %w", err)
+		}
+		return &con.ConsensusChannel, nil
+
+	} else {
+		con, err := NewLeaderChannel(ledger.FixedPart, turnNum, outcome, signatures)
```

Comment:
```suggestion
		con, err := NewFollowerChannel(ledger.FixedPart, turnNum, outcome, signatures)
```

Reply:
Why didn't this cause a failing test?

Reply:
A: the **only** difference in the resulting channel is that `myIndex` will be incorrect. The **only** place where that is actually used is when signing vars, which doesn't happen in the integration test.